uaclient/apt.py

```diff
@@ -391,3 +392,37 @@ def clean_apt_files(*, _entitlements=None):
 def get_installed_packages() -> "List[str]":
     out, _ = util.subp(["dpkg-query", "-W", "--showformat=${Package}\\n"])
     return out.splitlines()
+
+
+def setup_apt_proxy(
+    http_proxy: Optional[str] = None, https_proxy: Optional[str] = None
+) -> None:
+    """
+    Writes an apt conf file that configures apt to use the proxies provided as
+    args.
+    If both args are None, then no apt conf file is written. If this function
+    previously wrote a conf file, and was run again with both args as None,
+    the existing file is removed.
+
+    :param http_proxy: the url of the http proxy apt should use, or None
+    :param https_proxy: the url of the https proxy apt should use, or None
+    :return: None
+    """
+    apt_proxy_config = ""
+    if http_proxy is not None:
+        apt_proxy_config += status.MESSAGE_APT_PROXY_HTTP.format(
+            proxy_url=http_proxy
+        )
+    if https_proxy is not None:
+        apt_proxy_config += status.MESSAGE_APT_PROXY_HTTPS.format(
+            proxy_url=https_proxy
+        )
+    if apt_proxy_config != "":
+        apt_proxy_config = (
+            status.MESSAGE_APT_PROXY_CONFIG_HEADER + apt_proxy_config
+        )
+
+    if apt_proxy_config == "":
+        util.remove_file(APT_PROXY_CONF_FILE)
+    else:
+        util.write_file(APT_PROXY_CONF_FILE, apt_proxy_config)
```

Comment:
If we are about to write apt proxy, do we think it's worth inspecting whether apt already has a proxy config that we are going to collide with?
we could check         ["apt-config", "shell", "key", "Acquire::http::Proxy" ]                and if non-empty we know we may have a problem worth adding a prompt for.

Reply:
@blackboxsw I think we absolutely should do this, but do you mind if I break this piece into another PR to keep this one from growing much more?

Reply:
> @blackboxsw I think we absolutely should do this, but do you mind if I break this piece into another PR to keep this one from growing much more?

Absolutely. Separate PR is the way to go. I figure for any of these nits it could have been separated easily. None are blocking for this PR.